salt: update to 3007.1.

lvdh opened the pull request:
#### Testing the changes
- I tested the changes in this PR: **briefly**, on x86_64-glibc
  - The build installs successfully.
  - Running `salt-minion` is successful, including connection to an existing `salt-master`.
  - Running `salt-master` from this build is untested.

#### Local build testing
- I built this PR locally for my native architecture, **x86_64-glibc**
- I built this PR locally for these architectures, **all crossbuilds**:
  - aarch64
  - aarch64-musl
  - armv6l
  - armv6l-musl
  - armv7l
  - armv7l-musl
  - x86_64-musl

#### Additional notes
- `xlint template`: ok
- changes:
  - template:
    - Update version and checksum.
  - patches:
    - Remove one patch specific to version `3006.8`.
    - Adapt one patch to upstream updates, no functional changes.

lvdh commented:
Forgot to CC the maintainer, I do apologize: @Vaelatern

lvdh commented:
Ah, the suite contains a check for i686.
Glad to see it passed. I shall include it in future local crossbuilds.

Vaelatern commented:
We need to add more patches for the most recent python we are running.
https://patch-diff.githubusercontent.com/raw/saltstack/salt/pull/66899.patch is one

lvdh commented:
Thanks for the useful input! And for testing :) Looking forward to your findings, particularly the stuff that breaks.

I'll have a go at basic tests, this time with both minion+master on Void.

Vaelatern commented:
Yeah, I've tested and the patch I mentioned isn't enough yet...

Vaelatern commented:
Ok I got it functional for python 3.12 but 3.13 is on the horizon.

Vaelatern commented:
My branch is here https://github.com/Vaelatern/void-packages/tree/salt-3007

I want to test a master/minion setup with this too. Also want to test with python 3.13 and fix it up for that, in advance of the upcoming python3.13 bump @ahesford (I think you're doing that?)

Vaelatern commented:
Have you needed to make any changes to salt that I should be aware of so that I can incorporate Py 3.13 into salt?

ahesford commented:
No, salt builds as-is on my branch.

Vaelatern commented:
Confirming that this does not presently work with python 3.13. It was broken before, but there are more changes necessary to salt for the newest Python.

Vaelatern commented:
https://github.com/saltstack/salt/issues/67118

Vaelatern commented:
https://github.com/saltstack/salt/issues/67119

lvdh commented:
Couldn't have made that messier, if I tried. 🤷

@Vaelatern feel free to open a replacement PR from [your branch](https://github.com/Vaelatern/void-packages/tree/salt-3007). Surely it sees more progress than mine. I will definitely use/test it for rollouts I have coming up in the next weeks.

lvdh commented:
Thanks a bunch for your work @Vaelatern!
I picked up again using your branch. Added some new dependencies, after experimenting with a salt install from [source](https://github.com/saltstack/salt/tree/v3007.1):

- python3-MarkupSafe
- python3-aiohttp
- python3-cryptography
- python3-dateutil
- python3-importlib_metadata
- python3-openssl

At this point, salt-3007.1 successfully:
- builds, installs, runs
- configures master/minion communication
- responds to `test.*` queries
on fresh Voidlinux glibc installs.

However, `salt-call state.apply` of a simple state fails with error: `No Top file or master_tops data matches found. Please see master log for details.` Unfortunately, master log is empty.
This behavior is exactly what I experienced before on other Linux distributions, with salt 3007.1. At this time I'd assume it is not related to Void packaging.

For reference, `salt '*' test.version` output of the test systems:
```
Salt Version:
              Salt: 3007.1
     
    Python Version:
            Python: 3.13.1 (main, Dec 14 2024, 12:44:03) [GCC 13.2.0]
     
    Dependency Versions:
              cffi: 1.17.1
          cherrypy: 18.10.0
          dateutil: 2.9.0
         docker-py: Not Installed
             gitdb: Not Installed
         gitpython: Not Installed
            Jinja2: 3.1.5
           libgit2: Not Installed
      looseversion: 1.3.0
          M2Crypto: 0.42.0
              Mako: Not Installed
           msgpack: 1.1.0
      msgpack-pure: Not Installed
      mysql-python: Not Installed
         packaging: 24.2
         pycparser: 2.21
          pycrypto: Not Installed
      pycryptodome: 3.16.0
            pygit2: Not Installed
      python-gnupg: 0.5.1
            PyYAML: 6.0.2
             PyZMQ: 26.2.1
            relenv: Not Installed
             smmap: Not Installed
           timelib: 0.3.0
           Tornado: 6.4.2
               ZMQ: 4.3.5
     
    Salt Package Information:
      Package Type: Not Installed
     
    System Versions:
              dist: void rolling void
            locale: utf-8
           machine: x86_64
           release: 6.12.12_1
            system: Linux
           version: Void rolling void
```

Vaelatern commented:
```
salt-call state.apply of a simple state fails with error: No Top file or master_tops data matches found. Please see master log for details. Unfortunately, master log is empty.
```
Indeed, not a Void problem. I thought I'd patched this in my branch though.

Rebase your work repeatedly on upstream/master to iron out the multiple thousands of commits.

lvdh commented:
> Indeed, not a Void problem. I thought I'd patched this in my branch though.

Pulling in your branch added three commits, for new packages: `timelib`, `cherrypy`, `rpm-vercmp`.
Should these additions fix the `state.apply` error? Or is your patch unrelated to these?

> Rebase your work repeatedly on upstream/master to iron out the multiple thousands of commits. Merge commits on this repo are more trouble than they are worth.

Thank you for the heads-up. This branch is now cleaned up, as best as I could.
It did involve force-pushing, which I understood to be ok as per `CONTRIBUTING.md`. It has been a (long) while since I collaborated through git, feedback from experienced maintainers is definitely welcome. 🙏

> Good list of deps. I'm going to try to schedule some work on this in the coming days.

Whenever fits your schedule, of course. I appreciate the time you already spent on this.
This branch is now cleaned up and has both our changes, it should be 'usable' again.

Vaelatern commented:
Ok, my branch is now updated with the latest patches to test all the things.

I'll test it on one of my computers in the coming days, and merge if it works right. Patches include the runtime errors around the top file under python 3.11 up.

Vaelatern commented:
I've tested and it works right.
I've not tested a full setup just yet, but local execution definitely is working, which it doesn't (due to the bugs you've noted) from straight upstream.

2 patches that are proposed and in the salt issue tracker are proposed. I'll probably PR them against Salt and then merge here.

lvdh commented:
That makes most sense, for sure.
Thanks for following up. Looking forward to upstream accepting your patches.

Vaelatern commented:
I'm comfortable merging them here before salt upstream. Salt upstream is slow.

lvdh commented:
Ah, well... In that case, merge ahead, for sure. Looking forward to it.

Vaelatern commented:
Ok I've opened PRs against upstream. Please rebase your work against mine

lvdh commented:
This branch has been rebased against yours. And to the void-linux/void-packages:master, too.

In case the result differs from what you expect, I invited you as a collaborator on lvdh/void-packages.

Vaelatern commented:
Push access was helpful.

lvdh commented:
Happy to help. Thank you for all your work on this one! 🙏
